### DISCUSSION TITLE
Run useContractRead in a mapping error

### DISCUSSION THREAD

**Discussion by dev-cz** [wevm/wagmi#825] [Q&A]
Hi everyone, so I am using wagmi and am populating a map with that displays a list of contract adddresses. I am then trying to run a function that has a useContractRead in it with the address being a prop, so that for each address returned in the map, it calls the function with the userContractRead hook in it passing the address variable, and returning the data. But I keep getting very sporadic results. Sometimes itll get the data, and workd great. Other times it'll give me the "Rendered more hooks than during the previous render.

This is my function I am calling with variables.
```
`function GreenWallWinnings(divTrackerWin: string, address: string) {
  let greenWallContract = '0xe8356977Bb1c955B24CE2a1C620c961B2AeE8263'
  let wordContractChain = 3

  const { data, error } = useContractRead({
    addressOrName: greenWallContract,
    contractInterface: greenWallABI,
    chainId: wordContractChain,
    enabled: false,
    functionName: 'checkGreenWallWinnings',
    args: [divTrackerWin, address],
  })

  if (!data) {
    return '0'
  }

  if (data) {
    return data?.toString()
  }
}

export default GreenWallWinnings`
```

And I am calling it in another component like this.

`<div>{results?.divTrackerWin && getWinnings.call(results?.divTrackerWin, address)}</div>`

Sometimes it works, other times itll work when I refresh the page, then on certain wallets it doesnt work at all.

**Comment by leeburn** [comment]
I am having the same issue with a very similar function (i.e. return map of addresses then loop through addresses with useContractRead). 

My function works fine when the current map of addresses is no different to the previous map of addresses. However, if even a single new address is added, I get the "Rendered more hooks than during the previous render." error.

I can resolve the error by refreshing the page, but this is obviously not a useful solution.

I believe the problem is being caused because the current list of addresses is different to the previous list of addresses. 

Using the insight that a page refresh resolves the issue, I have attempted different ways to update the data and to ensure that the function is only reading from the updated data source, not the previous, but for the life of me I can't fix the problem.

Were you able to solve this?

**Reply by tmm** [reply]
Have some code to look at?

**Reply by leeburn** [reply]
Sure - see below. The purpose is to get the total balance of all addresses connected to the user's address. This code calls the list of addresses of the user and then loops through each address to get the address's balance, adding each to a running total. Thanks.

```
    const { data: ownedDETFs, isError, isLoading } = useContractRead({
        addressOrName: detfFactoryAddress[0],
        contractInterface: IPolybitDETFFactory,
        functionName: "getDETFAccounts",
        args: [walletOwner],
    })
    const [ownedDETFData, setOwnedDETFData] = useState(ownedDETFs)

    useEffect(() => {
        const data = ownedDETFs ? ownedDETFs : []
        setOwnedDETFData(data)
    }, [ownedDETFs])

    function GetBalanceOfDETF(detfAddress: string) {
        const { data, isError, isLoading, isSuccess } = useContractRead({
            addressOrName: detfAddress,
            contractInterface: IPolybitDETF,
            functionName: "getTotalBalanceInWeth"
        })
        const balance = data ? data : 0
        return balance
    }

    function GetTotalPortfolioWorth() {
        let totalBalance = 0
        ownedDETFData?.map((index: string) => {
            totalBalance = totalBalance + Number(GetBalanceOfDETF(index))
        })
        return totalBalance
    }

    let totalPortfolioWorth = GetTotalPortfolioWorth()
    console.log(totalPortfolioWorth)
```
